### PR TITLE
Feature/33 react testing library

### PR DESCRIPTION
**Pull request by mburri**
introduce (react-)testing-library to gradually replace Enzyme, see https://www.thoughtworks.com/radar/languages-and-frameworks/enzyme for reasons why

### CODE REVIEW

File: packages/spectrum/test/renderers/SpectrumAnyOfRenderer.test.tsx [hunk 1]
```diff
@@ -144,29 +122,32 @@ describe('Spectrum anyOf renderer', () => {
...
+    let tablist = getByRole('tablist');
+    let tabs = within(tablist).getAllByRole('tab');
+    let firstItem = tabs[0];
+    expect(firstItem).toHaveAttribute('aria-selected', 'true');
+    triggerPress(firstItem);
 
-    selectanyOfTab(wrapper, 1);
-    const nrOfRowsBeforeAdd = wrapper.find('tr');
-    clickAddButton(wrapper, 2);
-    const nrOfRowsAfterAdd = wrapper.find('tr');
```
Comment (hupf) [inline packages/spectrum/test/renderers/SpectrumAnyOfRenderer.test.tsx:161]: It's not the tab switching that is adding the mything, it's the click on the add button (or the '+'-Button, see `clickAddButton`). I suppose there should initially be one element present due to the given data/state.

File: packages/spectrum/test/renderers/SpectrumAnyOfRenderer.test.tsx [hunk 2]
```diff
@@ -215,30 +196,36 @@ describe('Spectrum anyOf renderer', () => {
...
+
+    expect(secondItem).toHaveAttribute('aria-selected', 'true');
+
+    const addButton = container.querySelector('button');
+    userEvent.click(addButton);
 
     await waitForAsync();
 
-    wrapper.update();
+    // todo: there is no input field available... what is missing here?
```
Comment (hupf) [inline packages/spectrum/test/renderers/SpectrumAnyOfRenderer.test.tsx:227]: Is there something missing in the uischema? May we have to look at the original test...

Or we haven't implemented a component that is used here.

File: packages/spectrum/test/util.tsx [hunk 3]
```diff
@@ -40,9 +40,40 @@ import {
...
+    </SpectrumThemeProvider>
+  );
+}
+
+// Triggers a "press" event on an element.
+export function triggerPress(element: any, opts = {}) {
+  fireEvent.mouseDown(element, { detail: 1, ...opts });
+  fireEvent.mouseUp(element, { detail: 1, ...opts });
+  fireEvent.click(element, { detail: 1, ...opts });
+}
```
Comment (hupf) [inline packages/spectrum/test/util.tsx:74]: :+1:

**Review by hupf — CHANGES_REQUESTED** [review]
Looks nice, not such hudge difference. Should be more or less easy to migrate the existing tests.

The "add" test should be fixable.

File: packages/spectrum/test/renderers/SpectrumAnyOfRenderer.test.tsx (hunk 1 above)
Comment (mburri) [reply packages/spectrum/test/renderers/SpectrumAnyOfRenderer.test.tsx:161]: ah - this got lost somehow

Comment (mburri) [reply packages/spectrum/test/renderers/SpectrumAnyOfRenderer.test.tsx:161]: fixed it

**Pull request merged**